src/engraving/dom/utils.cpp

```diff
@@ -1668,13 +1668,22 @@ bool segmentsAreAdjacentInRepeatStructure(const Segment* firstSeg, const Segment
     if (!firstSeg || !secondSeg) {
         return false;
     }
+    const MasterScore* master = firstSeg->masterScore();
+
     Measure* firstMeasure = firstSeg->measure();
     Measure* secondMeasure = secondSeg->measure();
 
     if (firstMeasure == secondMeasure) {
         return true;
     }
 
+    const MeasureBase* firstMasterMeasureBase = master->measure(firstMeasure->index());
+    const Measure* firstMasterMeasure = firstMasterMeasureBase
+                                        && firstMasterMeasureBase->isMeasure() ? toMeasure(firstMasterMeasureBase) : nullptr;
+    const MeasureBase* secondMasterMeasureBase = master->measure(secondMeasure->index());
+    const Measure* secondMasterMeasure = secondMasterMeasureBase
+                                         && secondMasterMeasureBase->isMeasure() ? toMeasure(secondMasterMeasureBase) : nullptr;
```

Comment:
Wouldn't this be more immediate by doing e.g. `master->tick2measure(firstMeasure->tick())`? It should also be much faster after your work on tick indexing

Reply:
Otherwise looks good, feel free to merge after this small change 👍

Reply:
Yes, definitely! I'd actually prefer to be overly cautious and make these changes in 4.6 as tick indexing won't be in the patch. This method of accessing measures has been fine for `findFollowing/PreviousRepeatMeasures` and I'd hate to break something now